### ISSUE TITLE
Add colorblind settings

### ISSUE THREAD

**Issue by epicEaston197** [Yellow-Dog-Man/Resonite-Issues#1947]
### Is your feature request related to a problem? Please describe.

Some users may have some color deficiencies that might make it difficult to use the platform and its interfaces especially proto flux

### Describe the solution you'd like

A variety of color blind settings to accommodate multiple variants of color blindness these would be easily readable with proto flux so creators can tailor their experiences to people's color deficiencies 

and if users are not comfortable with this they can turn on an additional privacy setting to hide this information then the setting will only affect official interfaces

### Describe alternatives you've considered

Maybe creating a cloud variable solution but this is not preferable as you have to be "in the know" to access these utilities

### Additional Context

This came up while I was having a Twitter conversation and they brought up a good point we should have color blind settings

### Requesters

@epicEaston197

**Comment by Frooxius** [comment]
What would the colorblind settings actually do?

Some of the things you mention would be better addressed by just making general changes - adding more patterns and shapes to things in addition to color to distinguish them - those are good as global changes, rather than a setting.

**Comment by epicEaston197** [comment]
> What would the colorblind settings actually do?
> 
> Some of the things you mention would be better addressed by just making general changes - adding more patterns and shapes to things in addition to color to distinguish them - those are good as global changes, rather than a setting.

I don't know what color blind settings actually do in games
I don't know how exactly they work or what they do but people appreciate them nonetheless

**Comment by Frooxius** [comment]
We'd need this to be more defined, because right now I'm not too sure what would actually be useful to make this more actionable.

Ideally getting feedback from people who have experience in this area or who are affected by colorblindness, on what would be most useful or what areas they struggle with.

**Comment by ko-tengu** [comment]
The most common and most appreciated colorblindness settings are things Resonite already does pretty well. We use distinct icons for all modes, we don't rely on color as anything other than a tertiary indicator, stuff like that.

The general full game colorblindness filters are generally disliked, and would be better served by making sure to reinforce to the community to try to follow Resonite's lead.

Microsoft has a good general set of guidelines to follow and refer to.
https://learn.microsoft.com/en-us/gaming/accessibility/guidelines

**Comment by shiftyscales** [comment]
@Frooxius - From what I've seen and heard of it in the past, the biggest contributing factor is just ensuring that any visual element that relies on colour in any way can also be distinguished in additional ways, e.g. use of icons, etc. to make things visually distinct even without the use of colour.

Likewise- one of the things to avoid would be the approach of adding a visual filter over the entire scene- this is an implementation I have heard complaints about.

This website/article is a good reference point for this particular issue.

https://gameaccessibilityguidelines.com/ensure-no-essential-information-is-conveyed-by-a-colour-alone/

Something I could see being potentially beneficial (?) would be if we were to let users locally edit the Resonite palette component which is used by all of the UI elements, etc. doing a one-to-one colour substitution for any colours the user has difficulty seeing.

If they could tune the colours themselves- it would allow them to pick colours that are sufficiently distinct/have sufficient contrast/meet their needs for their colour vision deficiency.

**Comment by ColinTimBarndt** [comment]
Things I'd like to see, if chromatic accessibility were to be considered:

- "high contrast" palette (like what Minecraft did)
- (Optional) configurable color palette, but I'd at least want to have default and high contrast presets
- Protoflux access to a user's accessibility settings ("has high contrast?" and maybe "color palette" nodes or drivers)

**Comment by ColinTimBarndt** [comment]
> Ideally getting feedback from people who have experience in this area or who are affected by colorblindness, on what would be most useful or what areas they struggle with.

For chromatically accessible game design, I highly recommend the following paper:

[Why Should Red and Green Never Be Seen? Exploring Color Blindness Simulations as Tools to Create Chromatically Accessible Games](https://dl.acm.org/doi/abs/10.1145/3611026)

- [On researchgate where you can request access from the authors](https://www.researchgate.net/publication/374465317_Why_Should_Red_and_Green_Never_Be_Seen_Exploring_Color_Blindness_Simulations_as_Tools_to_Create_Chromatically_Accessible_Games)

**Comment by epicEaston197** [comment]
I appreciate all of this information this will certainly help my main worry is protoflux wires the wires have so many different colors depending on what type you have there needs to be a nice way to distinguish types not only by their color but maybe by some icon or number or something

**Comment by ko-tengu** [comment]
Oh yeah, protoflux is probably somewhere that could use some help with that, you can get the types from pointing at the strip with a flux tool, but maybe making all the node cards a bit larger and having a representative icon or just write out the type is a good solution.

A super quick and dirty mockup I made of a potential improvement for both colorblind users and also just generally improving protoflux learnability
![Untitled](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/89536622/2c28c1ee-a87f-4013-b757-2842be12af3a)

**Comment by epicEaston197** [comment]
> Oh yeah, protoflux is probably somewhere that could use some help with that, you can get the types from pointing at the strip with a flux tool, but maybe making all the node cards a bit larger and having a representative icon or just write out the type is a good solution.
> 
> A super quick and dirty mockup I made of a potential improvement for both colorblind users and also just generally improving protoflux learnability

when you make the nodes black and white like that it is practically impossible to know the data types of these nodes there's no difference between an `int 1` and a `float 1` you're mock up definitely helps with this as seen the color is no longer necessary to know the data types of the inputs and outputs

**Comment by Frooxius** [comment]
@ko-tengu: The protoflux one is a bit more difficult.

The mockup only works for this particular node, because it doesn't have any outputs, but it will not work for lots of others, which already use the space you used for the type to have labels for the output connectors.

What we were wanting to do for types that I think would be better approach is to introduce unique black & white patterns for each common type, which would be displayed there.

-----------------------------

However more generally, I feel we should convert this into a discussion instead. E.g. a lot of the stuff proposed here aren't really settings and there's not a single actionable item from this.

Once we figure out what concrete changes and additions to make, we can make those into issues.

**Comment by epicEaston197** [comment]
>However more generally, I feel we should convert this into a discussion instead. E.g. a lot of the stuff proposed here aren't really settings and there's not a single actionable item from this.

@Frooxius yes please do this I agree with that this should be a discussion 
not sure if we can turn this issue into a discussion though i will start one if i need to

**Comment by epicEaston197** [comment]
okay I've created [Colorblind accessibility discussion #1959](https://github.com/Yellow-Dog-Man/Resonite-Issues/discussions/1959) all future discussion about this feature should be taken here

**Comment by ColinTimBarndt** [comment]
> A super quick and dirty mockup I made of a potential improvement for both colorblind users and also just generally improving protoflux learnability
> ![Untitled](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/89536622/2c28c1ee-a87f-4013-b757-2842be12af3a)

I'd recommend using color blindness simulations instead of simple black/white filters. Those issues can be partially improved by changing the palette depending on the dichromacy of the user. Monochromacy (black and white) is very rare while (Prot-/Deuteranomaly)[2] is fairly common, especially for male users. Additionally, a (second channel)[3] should be used (e.g. shapes of inputs or a texture on the wire).

Second channels are one recommendation made by the (Accessible Player Experiences (APX))[1] guidelines (also quoted by the paper I referred to).

[1]: https://accessible.games/accessible-player-experiences/
[2]: https://en.m.wikipedia.org/wiki/Color_blindness
[3]: https://accessible.games/accessible-player-experiences/access-patterns/second_channel/

**Comment by KierDran** [comment]
I think that the wires should be more texture oriented. Attached is an example of how a certain software does it... Much like the lasers being dotted in different patterns, the flux ribbons could be much more descriptive! (Especially for us extreme colorblind people)
![rtaImage](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/13511918/786d9c00-f22d-4660-9c05-8a01b0946b0b)

**Comment by epicEaston197** [comment]
> I think that the wires should be more texture oriented. Attached is an example of how a certain software does it... Much like the lasers being dotted in different patterns, the flux ribbons could be much more descriptive! (Especially for us extreme colorblind people)

great example but can you please move this to the discussion please [Colorblind accessibility discussion #1959](https://github.com/Yellow-Dog-Man/Resonite-Issues/discussions/1959)
